include/picongpu/particles/creation/moduleInterfaces/WriteOutSharedStateFunctor.hpp

```diff
@@ -0,0 +1,37 @@
+/* Copyright 2024 Brian Marre
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software you can redistribute it and or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+namespace picongpu::particles::creation::moduleInterfaces
+{
+    //! interface of WriteOutSharedStateFunctor
+    template<typename... T_KernelConfigOptions>
+    struct WriteOutSharedStateFunctor
+    {
+        template<typename T_Index, typename T_SharedStateType, typename... T_SharedDataBoxes>
+        HDINLINE static void write(
+            pmacc::DataSpace<picongpu::simDim> const superCellIndex,
+            T_Index const sharedDataBoxIndex,
+            T_SharedStateType const sharedState,
+            T_SharedDataBoxes... sharedDataBoxes);
```

Comment:
please rename, there is no need to say anything about implementation details e.g. that this object is shared between threads.

Reply:
as discussed offline renamed to `additionalData`